crates/uv-distribution-types/src/requires_python.rs

```diff
@@ -70,20 +70,14 @@ impl RequiresPython {
             .map(|specs| {
                 // Warn if there’s exactly one `~=` specifier without a patch.
                 if let [spec] = &specs[..] {
-                    if spec.is_tilde_without_patch() {
-                        if let Some((lo_b, hi_b)) = release_specifier_to_range(spec.clone())
-                            .bounding_range()
-                            .map(|(l, u)| (l.cloned(), u.cloned()))
-                        {
-                            let lo_spec = LowerBound::new(lo_b).specifier().unwrap();
-                            let hi_spec = UpperBound::new(hi_b).specifier().unwrap();
```

Comment:
I found this hard to follow and there's a lot of indirection that can be avoided for the case we care about.